domain/src/test/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/CalendarTest.java

```diff
@@ -626,4 +625,277 @@ public void createCalendarWithValidValuesShouldNotGenerateNotice() {
 
         assertTrue(buildResult.getData() instanceof Calendar);
     }
+
+    @Test
+    void disjointDatesShouldReturnFalseWhenCallingIsOverlapping() {
+        // periods `a` and `b` are totally disjoint
+        //  a |--------|            2020/01/01 -> 2020/01/31
+        //              b  |----|   2020/03/01 -> 2020/04/30
+        final Calendar.CalendarBuilder underTest = new Calendar.CalendarBuilder();
+
+        final Calendar firstCalendar = (Calendar) underTest.serviceId(SERVICE_ID)
+                .monday(0)
+                .tuesday(0)
+                .wednesday(0)
+                .thursday(0)
+                .friday(0)
+                .saturday(0)
+                .sunday(0)
+                .startDate(LocalDate.of(2020, 1, 1))
+                .endDate(LocalDate.of(2020, 1, 31))
+                .build()
+                .getData();
+
+        final Calendar nonOverlappingCalendar = (Calendar) underTest.serviceId(SERVICE_ID)
+                .monday(0)
+                .tuesday(0)
+                .wednesday(0)
+                .thursday(0)
+                .friday(0)
+                .saturday(0)
+                .sunday(0)
+                .startDate(LocalDate.of(2020, 3, 1))
+                .endDate(LocalDate.of(2020, 4, 30))
+                .build()
+                .getData();
+
+        assertFalse(firstCalendar.isOverlapping(nonOverlappingCalendar));
+        // permute periods `a` and `b`
+        //            b  |----|    2020/03/01 -> 2020/04/30
+        // a |---      -----|      2020/01/01 -> 2020/01/31
+        assertFalse(nonOverlappingCalendar.isOverlapping(firstCalendar));
+    }
+
+    @Test
+    void periodContainedOtherPeriodShouldReturnTrueWhenCallingIsOverlapping() {
+        // periods `b` is contained within period `a`
+        //  a |--------| 2020/01/01 -> 2020/12/31
+        //   b  |----|   2020/03/01 -> 2020/04/30
+        final Calendar.CalendarBuilder underTest = new Calendar.CalendarBuilder();
+
+        final Calendar firstCalendar = (Calendar) underTest.serviceId(SERVICE_ID)
```

Comment:
I'd suggest naming these variables like in the ASCII art, so it's easy to match the two. So this would be `...Calendar a...`.